src/runtime_src/core/common/info_platform.cpp

```diff
@@ -103,6 +103,50 @@ add_mig_info(const xrt_core::device* device, ptree_type& pt)
   }
 }
 
+static std::map<std::string, long long> 
+p2p_convert_config(xrt_core::query::p2p_config::result_type& config)
+{
+  std::map<std::string, long long> map;
+  for (auto& str : config) {
+    // str is in key:value format obtained from p2p_config query
+    auto pos = str.find(":");
+    std::string key = str.substr(0, pos);
+    try {
+      long long value = std::stoll(str.substr(pos + 1));
+      map[key] = value;
+    } catch (const std::exception&) {
+      // Failed to parse a non long long BAR value. Dont parse it into the map and move on!
+    }
+  }
+  return map;
+}
+
+void
+add_p2p_config(const xrt_core::device* device, ptree_type& pt)
+{
+  try {
+    ptree_type pt_p2p;
+    auto config = xrt_core::device_query<xq::p2p_config>(device);
+    auto config_map = p2p_convert_config(config);
+    for(auto& pair : config_map)
+      pt_p2p.add(pair.first, std::to_string(pair.second/1024/1024/1024)); // Turn bytes into GB
```

Comment:
Can you create a common function for this in XBUtilities like `bytes to string(uint64_t bytes)` which takes the value and convert to Byte, KB, MB, etc. I feel like we have such a function somewhere in the codebase since we parse memory sizes for the memory report. Maybe start looking there.

Reply:
One already existed as xrt_core::utils::unit_convert. Very handy!